src/Avalonia.X11/X11Window.cs

```diff
@@ -995,11 +995,11 @@ public void ShowDialog(IWindowImpl parent)
 
         public void SetIcon(IWindowIconImpl icon)
         {
-            var data = ((X11IconData)icon).Data;
+            var data = ((X11IconData)icon)?.Data;
```

Comment:
For `null` icons we should use `XDeleteProperty` instead.

Reply:
@kekekeks done, thanks!